eng/Versions.props

```diff
@@ -291,24 +292,21 @@
       create a test insertion in Visual Studio to validate.
     -->
     <NewtonsoftJsonVersion>13.0.1</NewtonsoftJsonVersion>
-    <StreamJsonRpcVersion>2.14.20</StreamJsonRpcVersion>
+    <StreamJsonRpcVersion>2.15.4-alpha</StreamJsonRpcVersion>
     <!--
       When updating the S.C.I or S.R.M version please let the MSBuild team know in advance so they
       can update to the same version. Version changes require a VS test insertion for validation.
     -->
-    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
-    <SystemReflectionMetadataVersion>6.0.1</SystemReflectionMetadataVersion>
-    <MicrosoftBclAsyncInterfacesVersion>6.0.0</MicrosoftBclAsyncInterfacesVersion>
+    <SystemCollectionsImmutableVersion>7.0.0</SystemCollectionsImmutableVersion>
+    <SystemReflectionMetadataVersion>7.0.0</SystemReflectionMetadataVersion>
+    <MicrosoftBclAsyncInterfacesVersion>7.0.0</MicrosoftBclAsyncInterfacesVersion>
+    <MicrosoftVisualStudioTelemetryVersion>17.6.23</MicrosoftVisualStudioTelemetryVersion>
   </PropertyGroup>
   <PropertyGroup Condition="$(OfficialBuild)">
     <RestoreSources>
       $(RestoreSources);
       https://pkgs.dev.azure.com/dnceng/internal/_packaging/dotnet-tools-internal/nuget/v3/index.json;
     </RestoreSources>
-    <MicrosoftVisualStudioTelemetryVersion>16.6.15</MicrosoftVisualStudioTelemetryVersion>
-  </PropertyGroup>
-  <PropertyGroup Condition="!$(OfficialBuild)">
-    <MicrosoftVisualStudioTelemetryVersion>16.6.9</MicrosoftVisualStudioTelemetryVersion>
```

Comment:
I don't know why you needed to split out the telemetry version by OfficialBuild, but it seems since the new version you require is newer than either of these, we might have an opportunity to simplify, so that's what I did here.

Reply:
@Cosifne @genlu? Looks like this might have been a point-in-time fallout when we had different stuff going into different branches.

Reply:
We do this because at that time 16.6.15 is not open to the public so only do that in internal build.
I remember @genlu has a PR remove this

Reply:
Oh it's not checked in yet https://github.com/dotnet/roslyn/pull/67032